x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1535,85 +1538,4 @@ private static LogicalPlan normalize(Aggregate aggregate, AttributeMap<Expressio
             return changed.get() ? new Aggregate(aggregate.source(), aggregate.child(), aggregate.groupings(), newAggs) : aggregate;
         }
     }
-
-    /**
-     * Replace aggregations that are duplicated inside an Aggregate with an Eval to avoid duplicated compute.
-     * stats a = min(x), b = min(x), c = count(*), d = count() by g
-     * becomes
-     * stats a = min(x), c = count(*) by g
-     * eval b = a, d = c
-     * keep a, b, c, d, g
-     */
-    static class ReplaceDuplicateAggWithEval extends OptimizerRules.OptimizerRule<Aggregate> {
```

Comment:
It was a leftover and not used anywhere - a previous PR incorporated it yet the code hanged around - I've used this opportunity to remove it instead of creating another PR to further validate that the check gets applied.